src/clj/game/core/turns.clj

```diff
@@ -150,9 +150,8 @@
   ([state] (make-eid state nil))
   ([state {:keys [source source-type]}]
    (merge {:eid (:eid (swap! state update-in [:eid] inc))}
-          (when source
-            {:source source
-             :source-type source-type}))))
+          (when source {:source source})
+          (when source-type {:source-type source-type}))))
```

Comment:
👏great👏change👏

Reply:
Yeah. I stumbled on that (and in the end didn't even need it) 😄